src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -18,10 +18,18 @@ case class ConstExpr(value: String) extends Expr
 case class FuncExpr(function: String, args: List[Expr], isAggregation: Boolean) extends Expr
 case class BinaryOpExpr(lhs: Expr, op: String, rhs: Expr) extends Expr
 case class TypecastExpr(lhs: Expr, rhs: String) extends Expr
+case class Placeholder extends Expr
+
+sealed trait Body
+case class Atom(name : String, terms : List[Expr]) extends Body
+case class ModifierAtom(modifier: AtomModifier, bodies: List[Body]) extends Body
```

Comment:
How about QuantifiedBody then since they roughly correspond to logical quantifiers?  I also don't like ModifiedSomething...